lib/segment/src/index/field_index/full_text_index/postings_iterator.rs

```diff
@@ -4,6 +4,7 @@ use super::posting_list::PostingList;
 
 pub fn intersect_postings_iterator<'a>(
     mut postings: Vec<&'a PostingList>,
+    filter: impl Fn(PointOffsetType) -> bool + 'a,
```

Comment:
In the immutable state, we don't delete anything from the posting list. Provide here the deleted filter which is empty for mutable state and deleted flag chacking for immutable one